apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1628,6 +1651,19 @@ func (u *UniterAPI) NetworkInfo(args params.NetworkInfoParams) (params.NetworkIn
 	if err != nil {
 		return params.NetworkInfoResults{}, err
 	}
+	defaultSpaceAddress, spaceAddressErr := machine.PrivateAddress()
+	if args.RelationId != nil {
+		rel, err := u.st.Relation(*args.RelationId)
+		if err != nil {
+			return params.NetworkInfoResults{}, err
+		}
+		ru, err := u.getRelationUnit(canAccess, rel.Tag().String(), unitTag)
+		if err != nil {
+			return params.NetworkInfoResults{}, err
+		}
+		defaultSpaceAddress, spaceAddressErr = ru.SettingsAddress()
```

Comment:
Perhaps a better name for the method would be "ExternalAddress". Since that's what we're really getting at here - what address to external callers use to contact this unit. Another option could be "IngressAddress"?
I called it defaultSpaceAddress because the GetNetworkInfoForSpaces method uses it in the network info result for the default space.